core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -447,6 +447,16 @@ public Location add(Operation operation) throws IOException {
         }
     }
 
+    /**
+     * The a {@linkplain Location} that will sort after the {@linkplain Location} returned by the last write but before any locations which
+     * can be returned by the next write.
+     */
+    public Location getLastWriteLocation() {
+        try (ReleasableLock lock = readLock.acquire()) {
+            return new Location(current.generation, current.sizeInBytes() - 1, Integer.MAX_VALUE);
```

Comment:
There are lots of choices that keep the sorting invariants. I picked Integer.MAX_VALUE because it is obviously fake. 0 would work if I returned `current.sizeInBytes()` without the `- 1` I think. So long as we never return a 0 sized location.